crates/pathfinder/src/core.rs

```diff
@@ -422,6 +423,8 @@ impl Chain {
             Chain::Testnet2 => StarkHash::from_u128(0x534e5f474f45524c4932),
             // SN_INTEGRATION
             Chain::Integration => StarkHash::from_u128(0x534E5F494E544547524154494F4E),
+            // SN_GOERLI
+            Chain::Custom => StarkHash::from_u128(0x534e5f474f45524c49u128),
```

Comment:
This will be something to make a parameter in a future version once StarkWare allows setting this and exposes it via the gateway or something.

Reply:
Hmm, so custom chains will now just use `SN_GOERLI`? That's pretty unfortunate.

Reply:
Currently its hardcoded into the cairo-vm even: https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/definitions/general_config.py#L40-L42

Nothing we can do about this currently as they don't expose this value anywhere.

Reply:
Hmm, not quite: you can set a custom chain ID in the Starknet OS config: https://github.com/starkware-libs/cairo-lang/blob/60233af8be7e444d1f53d41f622404dc47d5c00d/src/starkware/starknet/definitions/general_config.py#L79

So this is not an inherent limitation in the VM itself -- but maybe it is in the custom gateway code they have?